biojava-core/src/main/java/org/biojava/nbio/core/search/io/Result.java

```diff
@@ -0,0 +1,149 @@
+package org.biojava.nbio.core.search.io;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import org.biojava.nbio.core.sequence.template.Sequence;
+
+/**
+ * This class models a search result.
+ * You will find one of this for every query sequence specified in the run.
+ * 
+ * Designed by Paolo Pavan.
+ * You may want to find my contacts on Github and LinkedIn for code info 
+ * or discuss major changes.
+ * https://github.com/paolopavan
+ * 
+ * @author Paolo Pavan
+ */
+
+public abstract class Result implements Iterable<Hit>{
+    private String program;
+    private String version;
+    private String reference;
+    private String dbFile;
+
+    private HashMap<String,String> programSpecificParameters;
+    
+    private int iterationNumber;
+    private String queryID;
+    private String queryDef;
+    private int queryLength;
+    private Sequence querySequence;
+    private List<Hit> hits;
+    private int hitCounter = -1;
+
+    public Result(String program, String version, String reference, String dbFile, HashMap<String, String> programSpecificParameters, int iterationNumber, String queryID, String queryDef, int queryLength, List<Hit> hits, Sequence querySequence) {
+        this.program = program;
+        this.version = version;
+        this.reference = reference;
+        this.dbFile = dbFile;
+        this.programSpecificParameters = programSpecificParameters;
+        this.iterationNumber = iterationNumber;
+        this.queryID = queryID;
+        this.queryDef = queryDef;
+        this.queryLength = queryLength;
+        this.hits = hits;
+        this.querySequence = querySequence;
+    }
+    /**
+     * Experimental.
+     * Wants to return an hashcode designed to allow conceptual comparisons of search results.
+     * Fields unrelated to search are deliberately not considered.
+     * @return 
+     */
+    public int hashCode(){
+        String prefix = queryID+queryDef;
+        String suffix="";
+        if (hits != null){
+            for (Hit h: hits) {
+                String hashString = h.getHspsHashString();
+                if (hashString== null) return prefix.hashCode();
+                suffix += hashString;
+            }
+        }
+        return (prefix+suffix).hashCode();
+    }
+    
+    @Override
+    public boolean equals(Object o){
+        if (!(o instanceof Result)) return false;
+        
+        return o.hashCode() == this.hashCode();
+    }

```

Comment:
Same comments as above about hashCode and equals. Try using the eclipse built-in function to generate them.